src/api/ans.ts

```diff
@@ -0,0 +1,61 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+import { Account, AccountAddress } from "../core";
+import { getOwnerAddress, registerDomain } from "../internal/ans";
+import { InputSingleSignerTransaction, InputGenerateTransactionOptions } from "../transactions/types";
+import { HexInput } from "../types";
+import { AptosConfig } from "./aptosConfig";
+
+/**
+ * A class to handle all `ANS` operations
+ */
+export class ANS {
+  readonly config: AptosConfig;
+
+  constructor(config: AptosConfig) {
+    this.config = config;
+  }
+
+  /**
+   * Retrieve the owner address of a domain name or subdomain name.
+   *
+   * ```ts
+   * getOwnerAddress({domainName: "aptos", subdomainName: "test"})
+   * // Will return the owner address of "test.aptos.apt" or undefined
+   * ```
+   *
+   * @param args.domainName - A string of the domain name to retrieve
+   * @param args.subdomainName - A string of the subdomain name to retrieve
+   *
+   * @returns MoveAddressType if the name is owned, undefined otherwise
+   */
+  async getOwnerAddress(args: { domainName: string; subdomainName?: string }): Promise<AccountAddress | undefined> {
+    return getOwnerAddress({ aptosConfig: this.config, ...args });
+  }
+
+  /**
+   * Registers a new domain name
+   *
+   * @param args.sender - The sender account
+   * @param args.domainName - A string of the domain name to register, will be suffixed with .apt
+   * @param args.registrationDuration  - Time in seconds for how long the name will be registered for. This
+   * is part of the cost of the transaction. Must be a multiple of one year in seconds.
+   * @param args.targetAddress optional - The address the domain name will resolve to. If not provided,
+   * the sender's address will be used.
+   * @param args.toAddress optional - The address to send the domain name to. If not provided,
+   * the transaction will be sent to the router.
+   *
+   * @returns InputSingleSignerTransaction
+   */
+  async registerDomain(args: {
+    sender: Account;
+    domainName: string;
+    registrationDuration: 31536000;
```

Comment:
This is hardcoded?  What happens if it isn't this number? (basically seems strange to give a number as an input parameter if there isn't any choice)

Reply:
This is a quirk of the contract currently. The duration must be in increments of a year (in seconds). Currently the contract only accepts a single year, but will expand to multi-year registrations once they figure out their pricing model. 

When making these contracts my first impression was to stick close to the contracts, that way there would be consistency and developers wouldn't get confused if they were referencing the contract. After talking yesterday, perhaps it is better to favor simplicity. 

I could change this from inputting seconds to inputing years, and do the math for users.